src/hooks/use-tezos-token-activity.hook.ts

```diff
@@ -18,9 +20,19 @@ export const useTezosTokenActivity = (): UseActivityInterface => {
     async (lastId: number | null) => {
       const operations = await getTezosOperations(publicKeyHash, lastId);
 
-      setIsAllLoaded(operations.length === 0);
+      const filteredOperations = deduplicate<OperationInterface>(operations, (a, b) => a.hash === b.hash);
 
-      const loadedActivities = mapOperationsToActivities(publicKeyHash, operations);
+      setIsAllLoaded(filteredOperations.length === 0);
+
+      const operationGroups = (
+        await Promise.all(
+          filteredOperations.map(x =>
+            getOperationGroupByHash<OperationLiquidityBakingInterface>(x.hash).then(x => x.data)
+          )
+        )
+      ).flat();
+
+      const loadedActivities = mapOperationsToActivities(publicKeyHash, operationGroups);
       const activityGroups = transformActivityInterfaceToActivityGroups(loadedActivities);
```

Comment:
it makes no sense that `operationGroups` data is `flatted` and then is grouped by `hash`.  

look like you don't need `transformActivityInterfaceToActivityGroups` at all (you could make activity groups at the moment of fetching  `operationGroups`).